tests/nsrt_learning/strips_learning/test_gen_to_spec_learner.py

```diff
@@ -1001,3 +1006,154 @@ def test_multi_pass_backchaining(val):
+    goal2 = {
+        GroundAtom(pred, [])
+        for pred in rng.permutation([A, B, C, D, E])[:rng.integers(1, 5)]
+    }
+    s20 = State({dummy: list(rng.choice([0.0, 1.0], size=5))})
+    while True:
+        # Sample s21 until it is different from s20.
+        s21 = State({dummy: list(rng.choice([0.0, 1.0], size=5))})
+        # Force the goal to be achieved.
+        for atom in goal2:
+            s21.set(dummy, pred_to_feat[atom.predicate], 1.0)
+        if s21[dummy] != s20[dummy]:
+            break
+    if use_single_option:
+        acts = [act1]
+    else:
+        poss_acts = [[act1], [act2]]
+        acts = poss_acts[rng.integers(len(poss_acts))]
+    traj2 = LowLevelTrajectory([s20, s21], acts, True, 1)
+    task2 = Task(s20, goal2)
+
+    # Create trajectory 3 (length 2).
+    # Sample a goal.
+    goal3 = {
+        GroundAtom(pred, [])
+        for pred in rng.permutation([A, B, C, D, E])[:rng.integers(1, 5)]
+    }
+    s30 = State({dummy: list(rng.choice([0.0, 1.0], size=5))})
+    while True:
+        # Sample s31 until it is different from s30.
+        s31 = State({dummy: list(rng.choice([0.0, 1.0], size=5))})
+        # Force the goal to be achieved.
+        for atom in goal3:
+            s31.set(dummy, pred_to_feat[atom.predicate], 1.0)
+        if s31[dummy] != s30[dummy]:
+            break
+    if use_single_option:
+        acts = [act1]
+    else:
+        poss_acts = [[act1], [act2]]
+        acts = poss_acts[rng.integers(len(poss_acts))]
+    traj3 = LowLevelTrajectory([s30, s31], acts, True, 2)
+    task3 = Task(s30, goal3)
+
+    # Create trajectory 4 (length 3).
+    # Sample a goal.
+    goal4 = {
+        GroundAtom(pred, [])
+        for pred in rng.permutation([A, B, C, D, E])[:rng.integers(1, 5)]
+    }
+    s40 = State({dummy: list(rng.choice([0.0, 1.0], size=5))})
+    while True:
+        # Sample s41 until it is different from s40.
+        s41 = State({dummy: list(rng.choice([0.0, 1.0], size=5))})
+        if s41[dummy] != s40[dummy]:
+            break
+    while True:
+        # Sample s42 until it is different from s41.
+        s42 = State({dummy: list(rng.choice([0.0, 1.0], size=5))})
+        # Force the goal to be achieved.
+        for atom in goal4:
+            s42.set(dummy, pred_to_feat[atom.predicate], 1.0)
+        if s42[dummy] != s41[dummy]:
+            break
+    if use_single_option:
+        acts = [act1, act1]
+    else:
+        poss_acts = [[act1, act1], [act1, act2], [act2, act1], [act2, act2]]
+        acts = poss_acts[rng.integers(len(poss_acts))]
+    traj4 = LowLevelTrajectory([s40, s41, s42], acts, True, 3)
+    task4 = Task(s40, goal4)
+
+    trajs = [traj1, traj2, traj3, traj4][:num_demos]
+    tasks = [task1, task2, task3, task4][:num_demos]
+
+    ground_atom_trajs = utils.create_ground_atom_dataset(trajs, predicates)
+    segmented_trajs = [segment_trajectory(traj) for traj in ground_atom_trajs]
+
+    # Now, run the learner on the three demos.
```

Comment:
don't you mean 4 here?

Reply:
actually it depends on the `num_demos` parameter but ya will fix